test/e2e/gateway_api_test.go

```diff
@@ -276,6 +279,37 @@ func testGatewayAPIResourcesProtection(t *testing.T) {
 	}
 }
 
+// testOperatorDegradedCondition verifies that unmanaged (e.g. experimental)
+// Gateway API CRDs affect the ingress cluster operator's Degraded status.
+func testOperatorDegradedCondition(t *testing.T) {
+	// The experimental CRDs were created at the beginning of `TestGatewayAPI`.
+	// So the ingress cluster operator should report Degraded=True.
+	expectedDegraded := []configv1.ClusterOperatorStatusCondition{
+		{
+			Type:   configv1.OperatorDegraded,
+			Status: configv1.ConditionTrue,
+			Reason: "GatewayAPICRDsDegraded",
+		},
+	}
+	if err := waitForClusterOperatorConditions(t, kclient, expectedDegraded...); err != nil {
+		t.Errorf("Did not get expected degraded condition: %v", err)
+	}
+
+	// Remove the experimental CRDs to see the ingress cluster operator
+	// recover from the Degraded state.
+	bypassVAP(t, deleteExperimentalCRDs)
+
+	expectedNotDegraded := []configv1.ClusterOperatorStatusCondition{
+		{
+			Type:   configv1.OperatorDegraded,
+			Status: configv1.ConditionFalse,
+		},
+	}
```

Comment:
This will result in false positives when the clusteroperator is degraded for other reasons.  You might be better off doing a string search in the message string rather than checking the status condition's reason and status.

Reply:
https://github.com/openshift/cluster-ingress-operator/pull/1205#discussion_r2023724721.